eva_submission/eload_ingestion.py

```diff
@@ -411,30 +418,47 @@ def run_remap_and_cluster_workflow(self, target_assembly, resume):
 
-    def _get_target_assembly(self):
-        if self.taxonomy == 9606:
-            self.info('No remapping or clustering for human studies')
-            return None
+    def _get_supported_assembly_from_evapro(self, tax_id: int = None):
+        tax_id = tax_id or self.taxonomy
         with self.metadata_connection_handle as conn:
             current_query = (
-                f"SELECT assembly_id FROM evapro.supported_assembly_tracker "
-                f"WHERE taxonomy_id={self.taxonomy} AND current=true;"
+                f"SELECT assembly_id FROM {SUPPORTED_ASSEMBLY_TRACKER_TABLE}"
+                f"WHERE taxonomy_id={tax_id} AND current=true;"
             )
             results = get_all_results_for_query(conn, current_query)
             if len(results) > 0:
-                self.eload_cfg.set(self.config_section, 'remap_and_cluster', 'target_assembly', value=results[0][0])
                 return results[0][0]
-        self.warning(f'Could not find any current supported assembly for {self.taxonomy}, skipping clustering')
+        self.warning(f'Could not find any current supported assembly for {tax_id}, in EVAPRO!')
         return None
 
-    def _target_assembly_from_same_taxonomy(self, target_assembly):
-        # Find taxonomy of the target assembly
-        _, taxonomy_id_from_target = get_assembly_name_and_taxonomy_id(target_assembly)
-        if int(taxonomy_id_from_target) != int(self.taxonomy):
-            self.warning(f'Target assembly {target_assembly} is from a different taxonomy {taxonomy_id_from_target} '
-                         f'compared to the current project {self.taxonomy}. Therefore remapping will not be carried out!')
-            return False
-        return True
+    def _insert_new_supported_asm_from_ensembl(self, tax_id: int = None):
+        tax_id = tax_id or self.taxonomy
+        target_assembly = get_supported_asm_from_ensembl(tax_id)
+        if target_assembly:
+            add_to_supported_assemblies(self.metadata_connection_handle, source_of_assembly='Ensembl',
+                                        target_assembly=target_assembly, taxonomy_id=tax_id)
+        return target_assembly
+
+    def _get_target_assembly(self):
+        if self.taxonomy == 9606:
+            self.info('No remapping or clustering for human studies')
+            return None
+        target_assembly = self._get_supported_assembly_from_evapro() or self._insert_new_supported_asm_from_ensembl()
+        if target_assembly is None:
+            self.warning(f"Could not find remapping target assembly from EVAPRO or Ensembl for the submitted taxonomy: "
+                         f"{self.taxonomy}... Attempting to find assemblies in an alternate taxonomy...")
+            alt_tax_ids = set([tax_id for tax_id in
+                               [get_assembly_name_and_taxonomy_id(asm)[1] for asm in self._get_assembly_accessions()]
+                              if tax_id != self.taxonomy])
```

Comment:
```suggestion
            alt_tax_ids = {tax_id for tax_id in
                               [get_assembly_name_and_taxonomy_id(asm)[1] for asm in self._get_assembly_accessions()]
                              if tax_id != self.taxonomy}
```

Reply:
Also I wonder if it's worth making `_get_assembly_accessions()` a [`cached_property`](https://docs.python.org/3/library/functools.html#functools.cached_property) as we're now using it a number of times and I don't think it's necessary to recompute every time.

Reply:
I think it is retrieving the info from in-memory config since the config is loaded during [initialization](https://github.com/EBIvariation/eva-common-pyutils/blob/fc1012f3f30d53411a429d40fab9c68f6847babd/ebi_eva_common_pyutils/config.py#L16) anyway